layers/+spacemacs/spacemacs-layouts/funcs.el

```diff
@@ -608,3 +608,16 @@ FRAME defaults to the current frame."
                                         (eyebrowse--get 'last-slot frame))
                                   (get-frame-persp frame)
                                   frame))
+
+;; shell-pop integration
+
+(defun spacemacs//set-persp-specific-shell (arg)
+  "To be called after persp-switch. Make the default shell number to use equal to the layout number."
+  (progn
+    (loop for persp in (persp-names-current-frame-fast-ordered) and i from 1
+          when (string= persp arg)
+          do (progn (setq shell-pop-last-shell-buffer-index i) (return)))))
```

Comment:
Ah, I misread the code, you are right. 

That said, I wonder if it would be better to update the `make-shell-pop-command` macro to call `shell-pop` with the appropriate index (which is looked up by way of some persp name map.

I'm assuming this still has the problem with deleting or reordering perspectives causing the shells to get mismatched, or does that work now?